src/explain_lime_gradcam.py

```diff
@@ -109,128 +92,89 @@ def generate_affine_vals(num_imgs, config, im_size=224):
-            cam=cam,
-            preprocess_transform=preprocess_transform
-        )
+        pred_class, orig_mask = explain_img(i, inputs, "Identity", batch_predict, idx2label, target_class, config, cam=cam)
 
         # Get result for each of the transformed images
-        pos_ious = {k:None for k in config['transform_types']}
-        neg_ious = {k:None for k in config['transform_types']}
-        class_preds = {k:None for k in config['transform_types']}
         for t_type in config['transform_types']:
-            if t_type != "affine":
-                t_pred_class, pos_iou, neg_iou = get_result(
-                    i,
-                    img,
-                    # lime_img=lime_image,
-                    t_type=t_type,
-                    batch_predict=batch_predict,
-                    idx2label=idx2label,
-                    pill_transf=pill_transf,
-                    target_class=target_class,
-                    config=config,
-                    random_seed=config['random_seed'],
-                    to_save_imgs=config['to_save_imgs'],
-                    orig_mask=orig_mask,
-                    cam=random_cam if t_type == "random" else cam,
-                    preprocess_transform=preprocess_transform
-                )
-            else:
-                t_pred_class, pos_iou, neg_iou = get_result(
-                    i,
-                    img,
-                    # lime_img=lime_image,
-                    t_type=t_type,
-                    batch_predict=batch_predict,
-                    idx2label=idx2label,
-                    pill_transf=pill_transf,
-                    target_class=target_class,
-                    config=config,
-                    random_seed=config['random_seed'],
-                    to_save_imgs=config['to_save_imgs'],
-                    orig_mask=orig_mask,
-                    cam=cam,
-                    preprocess_transform=preprocess_transform,
-                    is_rot_only=is_rot_only,
-                    trans_val=trans_val,
-                    rot_val=rot_val,
-                    rot_val_deg=rot_val_deg,
-                    scale_val=scale_val,
-                )
-            pos_ious[t_type] = pos_iou
-            neg_ious[t_type] = neg_iou
-            class_preds[t_type] = t_pred_class
-
-        # Log results to csv file
-        results, results_df = save_and_update_results(
-            i,
-            results,
-            target_class,
-            pred_class,
-            pos_ious,
-            neg_ious,
-            class_preds,
-            config,
-            is_rot_only,
-            rot_val_deg,
-            trans_val,
-            scale_val,
-        )
+            logger.debug(f"Getting explanation for {t_type} image")
+            signed_mags = [0]
+
+            if t_type != 'random':
+                magnitudes, signs = get_dataset_augmentations(testset)[t_type]
+                signs = [1, -1] if signs else [1]
```

Comment:
the values returned by `get_dataset_augmentation` are a list and a bool. To avoid confusion (there are multiple `signs` variables), better to change the bool to `is_signed` (bool variables are usually prefixed by `is_` to denote that we expect a bool. Suggested change:
```suggestion
                magnitudes, is_signed = get_dataset_augmentations(testset)[t_type]
                signs = [1, -1] if is_signed else [1]
```

Reply:
Holy crud, github is letting me make a commit from this suggestion. You co-authored the commit!!! omg [a673c2f](https://github.com/rangeldaroya/cs670-project/pull/44/commits/a673c2fe50f8e99b3d721f029c0a181d1842259c) resolved.